merge/update.go

```diff
@@ -174,12 +179,19 @@ func (s *Updater) Apply(liveObject, configObject *typed.TypedValue, version fiel
 	if err != nil {
 		return nil, fieldpath.ManagedFields{}, fmt.Errorf("failed to get field set: %v", err)
 	}
+	if ignored != nil {
+		set = set.RecursiveDifference(ignored)
+		// TODO: is this correct. If we don't remove from lastSet pruning might remove the fields?
```

Comment:
After thinking about it again, this might not be required.
Pruning should remove fields from the object if they are not present in newObject and ignored.
Please confirm

Reply:
> lastSet.Set().RecursiveDifference(ignored)

I think that has to be wrong, we're not editing "VersionedSet" in place, and also `RecursiveDifference` returns a new object, so that's a no-op right now.

I would assume that the `lastSet` has been trimmed already (during the operation that set it), but it's possible that the ignored fields have changed since that manager was written, so doing it again seem reasonable.